File: backend/src/main/java/com/smarttravel/services/ReviewService.java
package com.smarttravel.services;

import com.smarttravel.dto.request.ReviewCreateRequest;
import com.smarttravel.dto.response.ReviewResponse;

import java.util.List;

public interface ReviewService {
    ReviewResponse createReview(ReviewCreateRequest request, String userEmail);
    List<ReviewResponse> getReviewsByTourId(Long tourId);
    boolean checkEligibility(Long tourId, String userEmail);
}

File: frontend/src/services/reviewService.ts
import axiosClient from './axiosClient';
import { Review, ReviewCreateRequest } from '../types/review';
import { ApiResponse } from '../types/common';

export const reviewService = {
  createReview: async (data: ReviewCreateRequest): Promise<ApiResponse<Review>> => {
    return await axiosClient.post('/reviews', data);
  },

  getReviewsByTourId: async (tourId: number): Promise<ApiResponse<Review[]>> => {
    return await axiosClient.get(`/reviews/tour/${tourId}`);
  },

  checkEligibility: async (tourId: number): Promise<ApiResponse<boolean>> => {
    return await axiosClient.get('/reviews/check-eligibility', { params: { tourId } });
  },
};
